storage/remote/read_handler_test.go

```diff
@@ -195,6 +197,103 @@ func BenchmarkStreamReadEndpoint(b *testing.B) {
 	}
 }
 
+type chunkedResponses []*prompb.ChunkedReadResponse
+
+var _ = (sort.Interface)(&chunkedResponses{})
+
+func (r chunkedResponses) Len() int {
+	return len(r)
+}
+
+func (r chunkedResponses) Less(i, j int) bool {
+	lblsi := map[string]string{}
+	for _, li := range r[i].ChunkedSeries[0].Labels {
+		lblsi[li.Name] = li.Value
+	}
+
+	lblsj := map[string]string{}
+	for _, lj := range r[j].ChunkedSeries[0].Labels {
+		lblsj[lj.Name] = lj.Value
+	}
+
+	return labels.Compare(labels.FromMap(lblsi), labels.FromMap(lblsj)) < 0
+}
+
+func (r chunkedResponses) Swap(i, j int) {
+	r[i], r[j] = r[j], r[i]
+}
+
+// Test for https://github.com/prometheus/prometheus/issues/12605.
+func TestRemoteReadSortedness(t *testing.T) {
```

Comment:
```suggestion
func TestRemoteReadSortedResult(t *testing.T) {
```

Reply:
Thanks!

Lgtm but perhaps it makes sense to merge this once we decide what's the expected result and ideally fix. Let's discuss the issue.